### PR TITLE
SCP-4267 SCP-4268 SCP-4269 Corrections to property-based tests

### PR DESCRIPTION
**Pull request by bwbush**
This contains three testing-related changes:
1. SCP-4267 allows the false-positives static-analysis test to timeout without failing.
2. SCP-4269 confirms the warning message for zero payment.
3. SCP-4268 fixes a logical error in the test for `getContinuation`.

Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

**Review by palas — APPROVED** [review]
Looks good.

I guess it would make more sense as a cabal variable, since it is pretty deployment specific (we could think a slower machine needs a longer timeout). But I wouldn't have said it if you hadn't asked, and we may want to have a default value if the variable is unset

Another thing is, I am not sure the timeout function will kill `z3` if it is running (it may remain zombie). There is a way of passing timeout values through SBV, but those are not guaranteed to work either https://hackage.haskell.org/package/sbv-9.0/docs/Data-SBV.html#v:setTimeOut.

**Comment by bwbush** [comment]
@palas, I will revise this to use a cabal variable.

A zombie `z3` won't impede the progression of testing, though that `z3` will use some resources on the VM where the test is performed. Since this error only happened about 2% of the time, I don't think it is worthwhile to dig deep in the process space to kill the `z3`.

**Comment by bwbush** [comment]
@palas, would you please re-review the cabal flag addition and two commits with minor testing-related fixes? Thanks.

File: marlowe/test/Spec/Marlowe/Semantics/Functions.hs [hunk 1]
```diff
@@ -882,11 +872,11 @@ checkGetContinuation =
...
            pure (input, case', contract, contractHash == contractHash')
     forAll' gen $ \(input, case', contract, hashesMatch) ->
-      _ALLOW_FAILED_CONTINUATION_ || case (input, case', getContinuation input case') of
-        (NormalInput{}    , Case{}          , contract') -> Just contract == contract'
-        (MerkleizedInput{}, MerkleizedCase{}, contract') -> (Just contract == contract') == hashesMatch
-        (_                , _               , Nothing  ) -> True
-        _                                                -> False
+      case (input, case', getContinuation input case') of
+        (NormalInput{}    , Case{}                                       , contract') -> Just contract == contract'
+        (MerkleizedInput _ contractHash _, MerkleizedCase _ contractHash', _        ) -> (contractHash == contractHash') == hashesMatch
```
Comment (palas) [inline marlowe/test/Spec/Marlowe/Semantics/Functions.hs:877]: Wait, isn't this shortcutting the test? Not part of the PR but just noticed
```suggestion
        (MerkleizedInput _ contractHash _, MerkleizedCase _ contractHash', _        ) -> contractHash == contractHash'
```

**Review by palas — APPROVED** [review]
Thank you. PR looks good, but I noticed something strange

File: marlowe/test/Spec/Marlowe/Semantics/Functions.hs (hunk 1 above)
Comment (bwbush) [reply marlowe/test/Spec/Marlowe/Semantics/Functions.hs]: Thanks for catching this! The combination of the somewhat convoluted generator and my trying to make the pattern matching too succinct end in making one of the matches tautological. This is fixed in 51fafdc82035fb096495f534907ea823171cc3e5, which is verbose and straightforward.
